lemonHope.py

```diff
@@ -14,6 +14,22 @@
 lemon = commands.Bot(command_prefix="Lemon, ")
 
 
+def with_lock(func):
+    """
+    Decorator to abstract our locking mechanism
+    Uses the standard asyncio.Lock()
+    """
+    async def doWithLock(*args, **kwargs):
```

Comment:
Now that I think about it, it might not work (my bad on original commit) - the locks are created independently inside threads and won't prevent each other from accessing resource. Maybe we should create a global lock for saveQuote and use it with ```async with save_quote_lock``` block?